libs/partners/nvidia-trt/langchain_nvidia_trt/llms.py

```diff
@@ -0,0 +1,404 @@
+
+    def _trim_batch_response(self, result_str: str) -> str:
+        """Trim batch response by removing prompt and extra generated text."""
+        # extract the generated part of the prompt
+        # TODO: This assumes llama-style prompting...
+        split = result_str.split("[/INST]", 1)
+        generated = split[-1]
+        end_token = generated.find("</s>")
+        if end_token == -1:
+            return generated
+        generated = generated[:end_token].strip()
+        return generated
+
+    def _send_stop_signals(self, model_name: str, request_id: str) -> None:
+        """Send the stop signal to the Triton Inference server."""
+        stop_inputs = self._generate_stop_signals()
+        self.client.async_stream_infer(
+            model_name,
+            stop_inputs,
+            request_id=request_id,
+            parameters={"Streaming": True},
+        )
+
+    def _generate_stop_signals(
+        self,
+    ) -> List[grpcclient.InferInput]:
+        """Generate the signal to stop the stream."""
+        inputs = [
+            grpcclient.InferInput("input_ids", [1, 1], "INT32"),
+            grpcclient.InferInput("input_lengths", [1, 1], "INT32"),
+            grpcclient.InferInput("request_output_len", [1, 1], "UINT32"),
+            grpcclient.InferInput("stop", [1, 1], "BOOL"),
+        ]
+        inputs[0].set_data_from_numpy(np.empty([1, 1], dtype=np.int32))
+        inputs[1].set_data_from_numpy(np.zeros([1, 1], dtype=np.int32))
+        inputs[2].set_data_from_numpy(np.array([[0]], dtype=np.uint32))
+        inputs[3].set_data_from_numpy(np.array([[True]], dtype="bool"))
+        return inputs
+
+    @staticmethod
+    def _process_result(result: Dict[str, str]) -> str:
+        """Post-process the result from the server."""
+
+        message = ModelInferResponse()
+        google.protobuf.json_format.Parse(json.dumps(result), message)
+        infer_result = grpcclient.InferResult(message)
+        np_res = infer_result.as_numpy("text_output")
+
+        generated_text = ""
+        if np_res is not None:
+            generated_text = "".join([token.decode() for token in np_res])
+
+        return generated_text
+
+    def _stream_callback(
+        self,
+        result_queue: queue.Queue[Union[Optional[Dict[str, str]], str]],
+        force_batch: bool,
+        result: grpcclient.InferResult,
+        error: str,
+        stop_words: List[str],
+    ) -> None:
+        """Add streamed result to queue."""
+        if error:
+            result_queue.put(error)
+        else:
+            response_raw: dict = result.get_response(as_json=True)
+            # TODO: Check the response is a map rather than a string
+            if "outputs" in response_raw:
+                # the very last response might have no output, just the final flag
+                response = self._process_result(response_raw)
+                if force_batch:
+                    response = self._trim_batch_response(response)
```

Comment:
For the streaming case, is it returning only new tokens? Or is it returning the full sequence on each iteration? If the former, this trim method will break (it looks for the input prompt/split token)